src/TTImp/Elab/Ambiguity.idr

```diff
@@ -49,7 +49,8 @@ expandAmbigName mode nest env orig args (IVar fc x) exp
                         fi <- fromIntegerName
                         si <- fromStringName
                         ci <- fromCharName
-                        let prims = mapMaybe id [fi, si, ci]
+                        di <- fromDoubleName
+                        let prims = mapMaybe id [fi, si, ci, di]
```

Comment:
This whole block could be refactored as a single call to `getPrimitiveNames`.

Reply:
I'm not sure this can be refactored as requested, as the values `fi`, `si` and so on are use further below in a call to `uniqueType`. But probably I just don't see the obvious.

Reply:
If we refactor `uniqType` too like I suggested then it's not an issue. :D

Reply:
The thing that confuses me is that `Core.Context.getPrimitiveNames` returns a list of `Name`s, but that's not what we need after refactoring `uniqType` the way you suggested. Anyway, I tried to clean this up (hopefully in accordance to what you had in mind). Sorry for being thick. :D

Reply:
Looks good!